sonar-text-plugin/src/main/resources/org/sonar/plugins/secrets/configuration/mysql.yaml

```diff
@@ -0,0 +1,287 @@
+        post:
+          # Avoid matching values found on SourceGraph that look like dummy passwords or insertions.
+          patternNot: "^.$|^(?i)pass(word|wd)?(?-i)$|^%s$|^\\$[A-Z_]+$|^\\{.*\\}$|^\\$[{(]"
+      examples:
+        - text: |
+            backend "mysql" {
+              conn_str = "mysql://user:pass@database.example/terraform_backend"
+            }
+          containsSecret: false
+        - text: |
+            connStr = fmt.Sprintf("mysql://%s:%s@%s:%s/%s?%s",
+              url.PathEscape(dbUser), url.PathEscape(dbPasswd), host, port, dbName, dbParam)
+          containsSecret: false
+        - text: |
+            if (!newConfig.DATABASE_URL) {
+              const encodedUser = encodeURIComponent(newConfig.POSTHOG_DB_USER)
+              const encodedPassword = encodeURIComponent(newConfig.POSTHOG_DB_PASSWORD)
+              newConfig.DATABASE_URL = `mysql://${encodedUser}:${encodedPassword}@${newConfig.POSTHOG_MYSQL_HOST}:${newConfig.POSTHOG_MYSQL_PORT}/${newConfig.POSTHOG_DB_NAME}`
+            }
+          containsSecret: false
+        - text: |
+            SqlSettingsDefaultDataSource = "mysql://mmuser:mostest@localhost/mattermost_test?connect_timeout=10"
+          containsSecret: true
+          match: mostest
+        - text: |
+            passwordFile := fs.String("password", "../../../tools/secrets/password.txt", "password file")
+            databasePrefix := fs.String("database-prefix", "mysql://mysql:mysql_password_padded_for_security@localhost:3306/ocr2vrf-test", "database prefix")
+            databaseSuffixes := fs.String("database-suffixes", "connect_timeout=10", "database parameters to be added")
+          containsSecret: true
+          match: mysql_password_padded_for_security
+        - text: |
+            driver: mysql
+            dsn: mysql://foouser:foopass@localhost:3306/testdb?connect_timeout=10
+            table: footable
+          containsSecret: true
+          match: foopass
+        - text: |
+            AIRFLOW_CONN_METADATA_DB=mysql+mysqldb://airflow:airflow@mysql:3306/airflow
+            AIRFLOW_VAR__METADATA_DB_SCHEMA=airflow
+          containsSecret: true
+          match: airflow
+
+    - id: mysql-env-mysql_password-unquoted
+      metadata:
+        name: MySQL database passwords should not be disclosed
+      detection:
+        pre:
+          include:
+            content:
+              - MYSQL_PASSWORD
+          reject:
+            ext:
+              - .md
+              - .html
+              - .example
+              - .template
+        matching:
+          pattern: "\\bMYSQL_PASSWORD=(?!\\\\?[\"'])([^\\s]+)(?:$|\\s)"
+        post:
+          # Avoid matching values found on SourceGraph that look like dummy passwords or insertions.
+          patternNot: "^(db|my)?_?pass(word|wd)?$|^None$|^\\*{3,}$|^\\$[A-Z_]+PASS(WORD)?$|^\\$[{(]|^`[a-z]+|^process\\.env\\.|^<[a-z ]{1,10}>?$"
```

Comment:
I have the feeling that this would make sense in most of the rules we write. It is used pretty much everywhere in your PR so:

- You could consider moving that at the provider level.
- We could move this to a common configuration. I don't think this is implemented for now tho.
- We could split this big pattern into multi sub things and specific heuristic post filtering module checks.

To clarify, this is just food for thought, I don't expect a change here.